firestore/incidents.py

```diff
@@ -303,29 +408,58 @@ def getAllIncidents(params, user_role):
     Fetches all incidents based on query parameters and user role.
     
     Args:
-    - params (dict): Query parameters
-    - user_role (str): The role of the user ('admin' or 'viewer')
+        params (dict): Query parameters including:
+            - start: Start datetime (default: 30 days ago)
+            - end: End datetime (default: now)
+            - state: State filter (optional)
+            - type: Incident type filter (optional)
+            - self_report_status: Status filter (optional)
+            - page_size: Results per page (default: 10, max: 100)
+            - cursor: Pagination cursor (optional)
+            - direction: Pagination direction (forward/backward)
+        user_role (str): User role ('admin' or 'viewer')
     
     Returns:
-    - (dict, int): Response and HTTP status code
+        dict: Response with incidents and pagination info
+        int: HTTP status code
     """
-    # Validate query parameters
     validation_error = get_all_validation(params, user_role)
     if validation_error:
         return validation_error
     
-    # Your existing logic to fetch incidents...
-    incidents = []  # Replace with actual fetching logic
-
+    # Set defaults
+    start = params.get('start', datetime.now() - timedelta(days=30))
+    end = params.get('end', datetime.now())
+    state = params.get('state', '')
+    incident_type = params.get('type', 'both')
+    self_report_status = params.get('self_report_status', 'approved' if user_role != 'admin' else 'all')
```

Comment:
why admin need to default to see all?
default should always be approved, even for admin.